server/src/HeapConnections.tsx

```diff
@@ -0,0 +1,134 @@
+import { Variable } from "./DapvizProvider";
+import { QuadraticBezierLine, QuadraticBezierLineRef } from "@react-three/drei";
+import React, { createContext, useMemo, useRef } from "react";
+import { useCallback, useState } from "react";
+import * as THREE from "three";
+import { useFrame } from "@react-three/fiber";
+
+
+export type HeapConnectionContextType = {
+  registerNode: (id: number, ref: React.RefObject<THREE.Group>) => void;
+  unregisterNode: (id: number) => void;
+};
+
+export const HeapConnectionContext = createContext<HeapConnectionContextType | null>(null);
+
+
+interface ConnectionLineProps {
+  parentRef: React.RefObject<THREE.Group>;
+  childRef: React.RefObject<THREE.Group>;
+}
+
+export const ConnectionLine = ({ parentRef, childRef }: ConnectionLineProps) => {
```

Comment:
dont export if you only use it in this file

```suggestion
const ConnectionLine = ({ parentRef, childRef }: ConnectionLineProps) => {
```